source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -611,7 +643,7 @@ JsonTranscoderFilter::encodeTrailers(Http::ResponseTrailerMap& trailers) {
 }
 
 void JsonTranscoderFilter::doTrailers(Http::ResponseHeaderOrTrailerMap& headers_or_trailers) {
-  if (error_ || !transcoder_) {
+  if (error_ || !transcoder_ || per_route_config_->disabled()) {
```

Comment:
is it possible that per_route_config_ is not initialized in this call?   e.g. decodeHeader() is not called. 

Same question for maybeConvertGrpcStatus() function